### DISCUSSION TITLE
Privacy policy regarding zed ai

### DISCUSSION THREAD

**Discussion by UnknownSuperficialNight** [zed-industries/zed#17625] [Feature Requests]
What data does Zed AI collect and transmit? For example, are prompts stored or collected? Are responses stored by either Zed or Anthropic?

**Comment by rikur** [comment]
I'd like to know this as well.

**Comment by 2zqa** [comment]
Check its privacy policy here:

https://zed.dev/terms#3353-zed-ai-services

"Other than as specified in Section 3.3.5.4 of these Terms, Zed will not use User Content for training of its models, or disclose User Content."

And in 3.3.5.4:

You may elect to share requests, responses and feedback comments (collectively "Model Improvement Feedback") with Zed, and Zed may use the same to improve Zed Edit Predictions models. You may opt-out of sharing Model Improvement Feedback at any time. [...] You may elect to share with Zed requests, responses and feedback regarding the Agent Panel and related Output (the “Agent Improvement Feedback”) with Zed, and Zed may use the same to improve the Agent Panel and related Output. Zed will only collect Agent Improvement Feedback when You elect to share the same.

**Reply by UnknownSuperficialNight** [reply]
This raises an important question: at what point does the Terms of Service (TOS) of Anthropic come into play? 

For example, if I interact with 'Zed AI,' which is hosted by Anthropic (To my understanding that is), the statement that 'Zed will not use User Content for training its models or disclose User Content' seems clear. However, while Zed itself may not store the data, does the interaction still fall under Anthropic's TOS?

In other words, even if Zed does not retain the prompt, the data is still transmitted to Anthropic, where it may be subject to their data storage and usage policies.

> The Zed Editor supports integration with API-based services maintained and operated by Zed (the "Zed AI Services").

So I could be wrong here, but I'm just asking for clarification as last I remember "Zed AI Services" was partnered or hosted with anthropic unless that changed, or I misinterpreted?

**Reply by 2zqa** [reply]
Good question. It seems they cover this as well in [Zeds Privacy Policy](https://zed.dev/privacy-policy#zero-data-retention-agreements):

"Zed has a zero-data retention agreement with Anthropic when using our [Zed AI Services](https://zed.dev/terms#3353-zed-ai-services)."

**Comment by macropin** [comment]
A privacy focused fork has been made: https://github.com/zedless-editor/zed

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
